lib/services/local/generic/generic.go

```diff
@@ -97,8 +97,8 @@ func NewService[T types.Resource](cfg *ServiceConfig[T]) (*Service[T], error) {
 }
 
 // GetResources returns a list of all resources.
-func (s *Service[T]) GetResources(ctx context.Context) ([]T, error) {
-	rangeStart := backend.Key(s.backendPrefix, "")
+func (s *Service[T]) GetResources(ctx context.Context, prefixParts ...string) ([]T, error) {
```

Comment:
MakeKey discussions aside, I'm finding this signature conceptually strange. Could you do a brief explanation of the key space we are dealing with and why this change is needed?

Reply:
So most of (but not all) backend prefixes have "single" namespaces, e.g. the `access_lists` prefix is expected to contain all of the access list objects.

For access list members, we have the backend prefix `access_list_members`, but then these can be further namespaces by access list name. So `access_list_members/accessList1` or something like that. The idea here being that in addition to being able to refer to resources by just the backend prefix, you can now append arbitrary prefixes in case you want to further namespace these objects.

Does that make sense?